bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java

```diff
@@ -328,4 +361,75 @@ private void refreshChannelSubscription(ChannelUID channelUID) {
     public Collection<Class<? extends ThingHandlerService>> getServices() {
         return Collections.singleton(LGWebOSActions.class);
     }
+
+    @Override
+    public String getUDN() {
+        return getLGWebOSConfig().getDeviceId();
+    }
+
+    @Override
+    public void onStatusChanged(boolean status) {
+        logger.debug("onStatusChanged status={}", status);
+        postUpdate(CHANNEL_POWER, status ? OnOffType.ON : OnOffType.OFF);
+    }
+
+    @Override
+    public void onValueReceived(@Nullable String variable, @Nullable String value, @Nullable String service) {
+    }
+
+    @Override
+    public void onServiceSubscribed(@Nullable String service, boolean succeeded) {
+    }
+
+    @Override
+    public void thingDiscovered(DiscoveryService source, DiscoveryResult result) {
+        if (THING_TYPE_WEBOSTV.equals(result.getThingTypeUID())) {
+            Map<String, Object> properties = result.getProperties();
+            String host = (String) properties.get(CONFIG_HOST);
+            if (host != null && host.equals(getLGWebOSConfig().getHost())) {
+                // Thing matching the discovery
+
+                // Update the thing properties from the discovery result
```

Comment:
so the media renderer will not have those additional properties I assume... ok, fine for me.

Reply:
If your thing is declared in a config file, it will have no properties set after startup. And it will not be set by the thing discovery.
Of course, the framework core code will check first if there are any change.

Reply:
And of course, we absolutely need to change (if necessary) the deviceId property.
This allows an automatic update of all existing things when you move from the old code to the new code.